client/components/categories/singleCategory.js

```diff
@@ -1,27 +1,39 @@
 import React, { Component } from 'react'
 import { connect } from 'react-redux'
 import { getCategoriesThunk } from '../../store/actionCreators'
+import { getProductsThunk } from '../../store/productThunks'
+import ProductCard from '../productCard'
 
 
 
 class SingleCategory extends Component {
 
   render() {
-    const { match: { params: { name } } } = this.props
+    console.log(this.props)
+    const { match: { params: { name, id } }, categories, products } = this.props
+    console.log(categories, products)
```

Comment:
try this.props.categories and this.props.products. i think this is why they are undefined.